### PR TITLE
Docs/adding first contrib

### PR DESCRIPTION
**Pull request by that-ar-guy**
## PR Summary
fixes : #3097 
Added a new file `first-contributors-guide.md` in `docs/license-contributing/ ` for guiding new contributors so that they can easily onboard to the project and start contributing.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/license-contributing/first-contributors-guide.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+8. Avoid using forced pushes or squashing changes while your PR is under review, as this makes it harder to review your changes.
+
+
+## 5. **Building and Testing Your Changes**
+
+Before opening a pull request, it’s important to build and test your changes locally. PSRule for Azure uses Continuous Integration (CI) pipelines to test changes across MacOS, Linux, and Windows configurations.
+
+1. Build and Test Locally:
+    - Ensure that you can build your changes locally. Follow the instructions in the [Building from Source](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/install.md#building-from-source) guide to set up your local environment for testing.
```
Comment (BernieWhite) [inline docs/license-contributing/first-contributors-guide.md:82]:
```suggestion
    - Ensure that you can build your changes locally. Follow the instructions in the [Building from Source](https://azure.github.io/PSRule.Rules.Azure/install/#building-from-source) guide to set up your local environment for testing.
```

File: docs/license-contributing/first-contributors-guide.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+2. Run All Tests:
+    - Before creating your PR, run the tests to make sure your changes are working as expected:
+      
+       ```
+       Invoke-Pester -Script .\tests
+       ```
+
```
Comment (BernieWhite) [inline docs/license-contributing/first-contributors-guide.md:90]: I think we are provide an example of a unit test for a rule here. Example is: https://github.com/Azure/psrule.rules.azure/blob/main/tests/PSRule.Rules.Azure.Tests/Azure.ACR.Tests.ps1#L40-L56

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @that-ar-guy for your PR. Great start. I've added some suggestions. 

Also you can add a link to the table of contents here (https://github.com/Azure/PSRule.Rules.Azure/blob/3e2a3b708c53ebcd029a2b5cb5437a150de2119c/mkdocs.yml#L77) below index so that it will appear in the docs.

**Review by BernieWhite — COMMENTED** [review]
Thanks for the updates @that-ar-guy. I've added some additional changes to your PR.

Have a look at let me know if you are happy with these changes.

**Comment by that-ar-guy** [comment]
@microsoft-github-policy-service agree

**Comment by that-ar-guy** [comment]
Thanks, @BernieWhite , for the updates! I've reviewed the additional changes, and they look great to me. I'm happy with these edits.

**Pull request merged**